### PR TITLE
Update to TestBench9 (JUnit5)

### PR DESCRIPTION
**Pull request by MarcinVaadin**
Updated to TestBench 9 including new tests that covers:
- JUnit5
- Selenium-Jupiter
- Parallel execution
- Retries on failure

Updated README.md.

### CODE REVIEW

File: src/test/java/com/vaadin/testbenchexample/junit5/WrongBrowserIT.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+/**
+ * In this example you can see how to filter test to run only on selected
+ * browser. {@code Capabilities} are being injected by TestBench.
+ */
+@RunLocally(Browser.CHROME) // the driver (could be multi-browser remote driver - Sauce)
+public class WrongBrowserIT extends AbstractJUnit5IT {
+
+    @BrowserTest // that test will be ignored
+    public void clickButton_notificationShown_safari() {
+        Assumptions.assumeTrue(BrowserUtil.isSafari(getCapabilities()),
```
Comment (mshabarov) [inline src/test/java/com/vaadin/testbenchexample/junit5/WrongBrowserIT.java:19]: Can we also show an example of injected `Capabilities` as a method's argument?

File: src/test/java/com/vaadin/testbenchexample/junit5/AbstractJUnit5IT.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        $(ButtonElement.class).id("button_" + op).click();
+        $(ButtonElement.class).id("button_" + b).click();
+        $(ButtonElement.class).id("button_=").click();
+        Assertions.assertEquals(result,
+                $(TextFieldElement.class).first().getValue());
+    }
+
+    @BeforeAll
+    public static void setupClass() {
+        WebDriverManager.chromedriver().setup();
```
Comment (mshabarov) [inline src/test/java/com/vaadin/testbenchexample/junit5/AbstractJUnit5IT.java:44]: Why is this needed? Can we embed it in the base class somehow, so that developers don't need to use this code in their tests?

Comment (MarcinVaadin) [reply src/test/java/com/vaadin/testbenchexample/junit5/AbstractJUnit5IT.java:44]: AbstractJUnit5IT - it is an abstract base class already. It is base for Chrome local tests.

File: src/test/java/com/vaadin/testbenchexample/junit5/WrongBrowserIT.java (hunk 1 above)
Comment (MarcinVaadin) [reply src/test/java/com/vaadin/testbenchexample/junit5/WrongBrowserIT.java:19]: Sure

File: src/test/java/com/vaadin/testbenchexample/junit5/AbstractJUnit5IT.java (hunk 2 above)
Comment (mshabarov) [reply src/test/java/com/vaadin/testbenchexample/junit5/AbstractJUnit5IT.java:44]: It is a base class for this demo project -> TestBench users would need to do the same in their projects, can we make their life easier?
And moreover, `WebDriverManager` is a part of `selenium-jupiter`, does the regular TestBench test need it?

Comment (mshabarov) [reply src/test/java/com/vaadin/testbenchexample/junit5/AbstractJUnit5IT.java:44]: Also noticed this code in the docs draft:
```
    @BeforeScenario
    public void setUpWebDriver() {
        WebDriverManager.chromedriver().setup();
        driver = TestBench.createDriver(new ChromeDriver());
        driver.get("http://" + IPAddress.findSiteLocalAddress() + ":8080");
        calculator = $(KeypadElement.class).first();
    }
```

It would be better if in `TestBench` method we would call to setup() automatically, or I miss something?

Comment (MarcinVaadin) [reply src/test/java/com/vaadin/testbenchexample/junit5/AbstractJUnit5IT.java:44]: I've updated code during work on docs, the point of BDD example is that it does not use `@BrowserTest` feature directly - it is run by JBehave making use of TestBench utility helpers.

Comment (mshabarov) [reply src/test/java/com/vaadin/testbenchexample/junit5/AbstractJUnit5IT.java:44]: Thanks for refactoring. I still have a question is `WebDriverManager.chromedriver().setup();` really needed for a regular TestBench test, since `selenium-jupiter` is optional to use, or I missed something?

Comment (MarcinVaadin) [reply src/test/java/com/vaadin/testbenchexample/junit5/AbstractJUnit5IT.java:44]: Example updated. Common parts extracted and now BDD tests are managed by TestBench.

File: pom.xml [hunk 3]
```diff
@@ -54,16 +54,22 @@
...
             <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>com.vaadin</groupId>
             <artifactId>vaadin-testbench</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>com.vaadin</groupId>
+            <artifactId>vaadin-testbench-core-junit5</artifactId>
```
Comment (mshabarov) [inline pom.xml:69]: Should it be `vaadin-testbench-junit5` now?

Comment (mshabarov) [reply pom.xml:69]: Leaving it as it now is fine, but let's remember to update the platform with latest testbench, wait for new platform release and update it here in this demo.

**Review by mshabarov — APPROVED** [review]

File: pom.xml (hunk 3 above)
Comment (MarcinVaadin) [reply pom.xml:69]: I think it is still not included in vaadin-bom

Comment (mshabarov) [reply pom.xml:69]: `vaadin-testbench-junit5` is not released in stable version, only available in SNAPSHOT.
Here https://tools.vaadin.com/nexus/content/repositories/vaadin-prereleases/com/vaadin/vaadin-testbench-junit5/24.0-SNAPSHOT/ .

But after platform next release, we can use it.
`vaadin-testbench-junit5` dependes on `vaadin-bom`
`vaadin-bom` contains `vaadin-testbench-bom`
`vaadin-testbench-bom` includes `vaadin-testbench-core-junit5`.

So everything should work.
